crowbar_framework/app/models/deployer_service.rb

```diff
@@ -138,4 +138,20 @@ def transition(inst, name, state)
     Rails.logger.debug("Deployer transition: exiting: #{name} for #{state}")
     return [200, { name: name }]
   end
+
+  # try to know if we can skip a node from running chef-client
+  def skip_unchanged_node?(node_name, old_role, new_role)
+    # if old_role is nil, then we are applying the barclamp for the first time
+    return false if old_role.nil?
+
+    # if the node changed roles, then we need to apply
+    return false if node_changed_roles?(node_name, old_role, new_role)
+
+    # There is not need to check attributes because they aren't necesarry for the current barclamp
```

Comment:
Typo: "necessary"

Also, I would rephrase to "they are not used in cookbooks"; they are actually used in the barclamp (in the rails application).